<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 General Platforms | Data Management for Cancer Research</title>
  <meta name="description" content="This course covers material about economic, privacy, security, usability, and discoverability aspects of data management for cancer research." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 General Platforms | Data Management for Cancer Research" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This course covers material about economic, privacy, security, usability, and discoverability aspects of data management for cancer research." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 General Platforms | Data Management for Cancer Research" />
  
  <meta name="twitter:description" content="This course covers material about economic, privacy, security, usability, and discoverability aspects of data management for cancer research." />
  



<meta name="date" content="2021-11-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/ITN_favicon.ico" type="image/x-icon" />
<link rel="prev" href="shared-computing-etiquette.html"/>
<link rel="next" href="specific-platforms.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.5.0/jquery-3.5.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style_ITN.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://www.itcrtraining.org/"><img src="assets/ITN_logo.png" style="padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this course</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="computing-basics.html"><a href="computing-basics.html"><i class="fa fa-check"></i><b>2</b> Computing basics</a>
<ul>
<li class="chapter" data-level="2.1" data-path="computing-basics.html"><a href="computing-basics.html#computing-components"><i class="fa fa-check"></i><b>2.1</b> Computing Components</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="computing-basics.html"><a href="computing-basics.html#transistors"><i class="fa fa-check"></i><b>2.1.1</b> Transistors</a></li>
<li class="chapter" data-level="2.1.2" data-path="computing-basics.html"><a href="computing-basics.html#alu---arithmetic-logic-unit"><i class="fa fa-check"></i><b>2.1.2</b> ALU - Arithmetic Logic Unit</a></li>
<li class="chapter" data-level="2.1.3" data-path="computing-basics.html"><a href="computing-basics.html#binary-data"><i class="fa fa-check"></i><b>2.1.3</b> Binary data</a></li>
<li class="chapter" data-level="2.1.4" data-path="computing-basics.html"><a href="computing-basics.html#flip-flops-and-registers"><i class="fa fa-check"></i><b>2.1.4</b> Flip-flops and registers</a></li>
<li class="chapter" data-level="2.1.5" data-path="computing-basics.html"><a href="computing-basics.html#cpu---central-processing-unit"><i class="fa fa-check"></i><b>2.1.5</b> <strong>CPU</strong> - Central Processing Unit</a></li>
<li class="chapter" data-level="2.1.6" data-path="computing-basics.html"><a href="computing-basics.html#memory-or-ram---short-term-memory"><i class="fa fa-check"></i><b>2.1.6</b> <strong>Memory or RAM</strong> - short-term memory</a></li>
<li class="chapter" data-level="2.1.7" data-path="computing-basics.html"><a href="computing-basics.html#storage---long-term-memory"><i class="fa fa-check"></i><b>2.1.7</b> <strong>Storage</strong> - long-term memory</a></li>
<li class="chapter" data-level="2.1.8" data-path="computing-basics.html"><a href="computing-basics.html#hardware-and-software"><i class="fa fa-check"></i><b>2.1.8</b> Hardware and software</a></li>
<li class="chapter" data-level="2.1.9" data-path="computing-basics.html"><a href="computing-basics.html#operating-systems"><i class="fa fa-check"></i><b>2.1.9</b> Operating systems</a></li>
<li class="chapter" data-level="2.1.10" data-path="computing-basics.html"><a href="computing-basics.html#historical-context"><i class="fa fa-check"></i><b>2.1.10</b> Historical context</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="computing-basics.html"><a href="computing-basics.html#conclusion"><i class="fa fa-check"></i><b>2.2</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="computing-systems.html"><a href="computing-systems.html"><i class="fa fa-check"></i><b>3</b> Computing Systems</a>
<ul>
<li class="chapter" data-level="3.1" data-path="computing-systems.html"><a href="computing-systems.html#data-sizes"><i class="fa fa-check"></i><b>3.1</b> Data Sizes</a></li>
<li class="chapter" data-level="3.2" data-path="computing-systems.html"><a href="computing-systems.html#computing-capacity"><i class="fa fa-check"></i><b>3.2</b> Computing Capacity</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="computing-systems.html"><a href="computing-systems.html#checking-your-computer-capacity---mac"><i class="fa fa-check"></i><b>3.2.1</b> Checking your computer capacity - Mac</a></li>
<li class="chapter" data-level="3.2.2" data-path="computing-systems.html"><a href="computing-systems.html#checking-your-computer-capacity---windowspc"><i class="fa fa-check"></i><b>3.2.2</b> Checking your computer capacity - Windows/PC</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="computing-systems.html"><a href="computing-systems.html#file-sizes"><i class="fa fa-check"></i><b>3.3</b> File Sizes</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="computing-systems.html"><a href="computing-systems.html#genomic-data-file-sizes"><i class="fa fa-check"></i><b>3.3.1</b> Genomic data file sizes</a></li>
<li class="chapter" data-level="3.3.2" data-path="computing-systems.html"><a href="computing-systems.html#imaging-data-file-sizes"><i class="fa fa-check"></i><b>3.3.2</b> Imaging Data File Sizes</a></li>
<li class="chapter" data-level="3.3.3" data-path="computing-systems.html"><a href="computing-systems.html#clinical-data-file-sizes"><i class="fa fa-check"></i><b>3.3.3</b> Clinical Data File Sizes</a></li>
<li class="chapter" data-level="3.3.4" data-path="computing-systems.html"><a href="computing-systems.html#checking-file-sizes-on-mac"><i class="fa fa-check"></i><b>3.3.4</b> Checking file sizes on Mac</a></li>
<li class="chapter" data-level="3.3.5" data-path="computing-systems.html"><a href="computing-systems.html#checking-file-sizes-on-pcwindows"><i class="fa fa-check"></i><b>3.3.5</b> Checking file sizes on PC/Windows</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="computing-systems.html"><a href="computing-systems.html#computing-options"><i class="fa fa-check"></i><b>3.4</b> Computing Options</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="computing-systems.html"><a href="computing-systems.html#personal-computers"><i class="fa fa-check"></i><b>3.4.1</b> <strong>Personal computers</strong></a></li>
<li class="chapter" data-level="3.4.2" data-path="computing-systems.html"><a href="computing-systems.html#shared-computers-or-servers"><i class="fa fa-check"></i><b>3.4.2</b> <strong>Shared Computers or Servers</strong></a></li>
<li class="chapter" data-level="3.4.3" data-path="computing-systems.html"><a href="computing-systems.html#computer-cluster"><i class="fa fa-check"></i><b>3.4.3</b> <span>Computer Cluster</span></a></li>
<li class="chapter" data-level="3.4.4" data-path="computing-systems.html"><a href="computing-systems.html#computer-grid"><i class="fa fa-check"></i><b>3.4.4</b> <span>Computer Grid</span></a></li>
<li class="chapter" data-level="3.4.5" data-path="computing-systems.html"><a href="computing-systems.html#accessing-shared-computer-resources"><i class="fa fa-check"></i><b>3.4.5</b> Accessing Shared Computer Resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html"><i class="fa fa-check"></i><b>4</b> Shared Computing Etiquette</a>
<ul>
<li class="chapter" data-level="4.1" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#general-guidelines-for-shared-computing-resources"><i class="fa fa-check"></i><b>4.1</b> General Guidelines for shared computing resources</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#security-guidelines"><i class="fa fa-check"></i><b>4.1.1</b> Security guidelines</a></li>
<li class="chapter" data-level="4.1.2" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#overall-use-guidelines"><i class="fa fa-check"></i><b>4.1.2</b> Overall use guidelines</a></li>
<li class="chapter" data-level="4.1.3" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#daily-use-guidelines"><i class="fa fa-check"></i><b>4.1.3</b> Daily use guidelines</a></li>
<li class="chapter" data-level="4.1.4" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#communication-guidelines"><i class="fa fa-check"></i><b>4.1.4</b> Communication Guidelines</a></li>
<li class="chapter" data-level="4.1.5" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#specific-rules"><i class="fa fa-check"></i><b>4.1.5</b> Specific Rules</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#interacting-with-shared-resources"><i class="fa fa-check"></i><b>4.2</b> Interacting with shared resources</a></li>
<li class="chapter" data-level="4.3" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#running-jobs"><i class="fa fa-check"></i><b>4.3</b> Running Jobs</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#specifying-memory-ram-needs"><i class="fa fa-check"></i><b>4.3.1</b> Specifying memory (RAM) needs</a></li>
<li class="chapter" data-level="4.3.2" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#checking-status"><i class="fa fa-check"></i><b>4.3.2</b> Checking status</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="shared-computing-etiquette.html"><a href="shared-computing-etiquette.html#storage"><i class="fa fa-check"></i><b>4.4</b> Storage</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="general-platforms.html"><a href="general-platforms.html"><i class="fa fa-check"></i><b>5</b> General Platforms</a>
<ul>
<li class="chapter" data-level="5.0.1" data-path="general-platforms.html"><a href="general-platforms.html#galaxy"><i class="fa fa-check"></i><b>5.0.1</b> Galaxy</a></li>
<li class="chapter" data-level="5.0.2" data-path="general-platforms.html"><a href="general-platforms.html#cyverse"><i class="fa fa-check"></i><b>5.0.2</b> CyVerse</a></li>
<li class="chapter" data-level="5.0.3" data-path="general-platforms.html"><a href="general-platforms.html#sciserver"><i class="fa fa-check"></i><b>5.0.3</b> SciServer</a></li>
<li class="chapter" data-level="5.0.4" data-path="general-platforms.html"><a href="general-platforms.html#materials-cloud"><i class="fa fa-check"></i><b>5.0.4</b> Materials Cloud</a></li>
<li class="chapter" data-level="5.0.5" data-path="general-platforms.html"><a href="general-platforms.html#atlas.ti"><i class="fa fa-check"></i><b>5.0.5</b> ATLAS.ti</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="specific-platforms.html"><a href="specific-platforms.html"><i class="fa fa-check"></i><b>6</b> Specific Platforms</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="specific-platforms.html"><a href="specific-platforms.html#imaging-data-analysis-platforms"><i class="fa fa-check"></i><b>6.0.1</b> Imaging Data Analysis Platforms</a></li>
<li class="chapter" data-level="6.0.2" data-path="specific-platforms.html"><a href="specific-platforms.html#quip"><i class="fa fa-check"></i><b>6.0.2</b> QuIP</a></li>
<li class="chapter" data-level="6.0.3" data-path="specific-platforms.html"><a href="specific-platforms.html#genomic-data-analysis-platforms"><i class="fa fa-check"></i><b>6.0.3</b> Genomic Data Analysis Platforms</a></li>
<li class="chapter" data-level="6.0.4" data-path="specific-platforms.html"><a href="specific-platforms.html#section"><i class="fa fa-check"></i><b>6.0.4</b> </a></li>
<li class="chapter" data-level="6.0.5" data-path="specific-platforms.html"><a href="specific-platforms.html#clinical-data-analysis-platforms"><i class="fa fa-check"></i><b>6.0.5</b> Clinical Data Analysis Platforms</a></li>
<li class="chapter" data-level="6.1" data-path="specific-platforms.html"><a href="specific-platforms.html#wuflux"><i class="fa fa-check"></i><b>6.1</b> WUFlux</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="specific-platforms.html"><a href="specific-platforms.html#machine-learning"><i class="fa fa-check"></i><b>6.1.1</b> Machine Learning…</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="data-management-decisions.html"><a href="data-management-decisions.html"><i class="fa fa-check"></i><b>7</b> Data Management Decisions</a>
<ul>
<li class="chapter" data-level="7.1" data-path="data-management-decisions.html"><a href="data-management-decisions.html#computing-platforms"><i class="fa fa-check"></i><b>7.1</b> Computing Platforms</a></li>
<li class="chapter" data-level="7.2" data-path="data-management-decisions.html"><a href="data-management-decisions.html#choosing-a-computing-platform"><i class="fa fa-check"></i><b>7.2</b> Choosing a Computing Platform</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="data-management-decisions.html"><a href="data-management-decisions.html#important-questions"><i class="fa fa-check"></i><b>7.2.1</b> Important questions</a></li>
<li class="chapter" data-level="7.2.2" data-path="data-management-decisions.html"><a href="data-management-decisions.html#benefits-of-cloud-computing"><i class="fa fa-check"></i><b>7.2.2</b> Benefits of Cloud Computing</a></li>
<li class="chapter" data-level="7.2.3" data-path="data-management-decisions.html"><a href="data-management-decisions.html#challenges-of-cloud-computing"><i class="fa fa-check"></i><b>7.2.3</b> Challenges of Cloud Computing</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="data-management-decisions.html"><a href="data-management-decisions.html#local-costs"><i class="fa fa-check"></i><b>7.3</b> Local Costs</a></li>
<li class="chapter" data-level="7.4" data-path="data-management-decisions.html"><a href="data-management-decisions.html#cloud-costs"><i class="fa fa-check"></i><b>7.4</b> Cloud Costs</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="data-security.html"><a href="data-security.html"><i class="fa fa-check"></i><b>8</b> Data Security</a></li>
<li class="chapter" data-level="9" data-path="data-privacy.html"><a href="data-privacy.html"><i class="fa fa-check"></i><b>9</b> Data Privacy</a></li>
<li class="chapter" data-level="10" data-path="data-sharing.html"><a href="data-sharing.html"><i class="fa fa-check"></i><b>10</b> Data Sharing</a>
<ul>
<li class="chapter" data-level="10.1" data-path="data-sharing.html"><a href="data-sharing.html#data-sharing-is-important"><i class="fa fa-check"></i><b>10.1</b> Data sharing is important!</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="data-ethics.html"><a href="data-ethics.html"><i class="fa fa-check"></i><b>11</b> Data Ethics</a></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/Data_Management" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; filter: grayscale(100%); padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
<p style="padding-left: 40px;"><div class="trapezoid" style = "padding-left: 40px;"><span>  <a href="https://forms.gle/hc8Xt3Y2Znjb6M4Y7"> Click here to provide feedback</a> <img src="assets/itcr_arrow.png" style=" width: 10%" ></span></div></p>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Management for Cancer Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
  <html>
  
  <head>
  <title>Title</title>
  </head>
  
  <body>
  
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-TGSJ7H9TXE"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    
    gtag('config', 'G-TGSJ7H9TXE');
  </script>
      
  </body>
  </html>
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=1.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
  <link href="assets/style_ITN.css" rel="stylesheet">
</head>
        


<div class="hero-image-container"> 
  <img class= "hero-image" src="assets/itcr_main_image.png">
</div>
<div id="general-platforms" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> General Platforms</h1>
<p>In this chapter we will provide examples of data management systems that you might find useful for your research. Please note that we aim to provide a general overview of options and thus this is not a complete list. Let us know if there is a platform or system that you think we should include!</p>
<p><img src="resources/images/05-General_Platforms_files/figure-html/1B4LwuvgA6aUopOHEAbES1Agjy7Ex2IpVAoUIoBFbsq0_gf5f8818810_1_171.png" title="Learning Objectives: 1.Recognize general informatics computing and data management platforms. 2.Be aware of specific genomics data management systems 3.Be aware of specific imaging data management systems  4.Be aware of specific clinical data management systems" alt="Learning Objectives: 1.Recognize general informatics computing and data management platforms. 2.Be aware of specific genomics data management systems 3.Be aware of specific imaging data management systems  4.Be aware of specific clinical data management systems" width="100%" style="display: block; margin: auto;" /></p>
<p>We will start out with more general computing platforms, that allow you flexibility to perform analyses with a variety of types of data.</p>
<div id="galaxy" class="section level3" number="5.0.1">
<h3><span class="header-section-number">5.0.1</span> Galaxy</h3>
<p>This section was written by <a href="https://goeckslab.org/people/jeremy.html">Jeremy Goecks</a>:</p>
<p>Galaxy is a web-based computational workbench that connects analysis tools, biomedical datasets, computing resources, a graphical user interface, and a programmatic API. Galaxy (<a href="https://galaxyproject.org/" class="uri">https://galaxyproject.org/</a>) enables accessible, reproducible, and collaborative biomedical data science by anyone regardless of their informatics expertise. There are more than 8,000 analysis tools and 200 visualizations integrated into Galaxy that can be used to process a wide variety of biomedical datasets. This includes tools for analyzing genomic, transcriptomic (RNA-seq), proteomic, metabolomic, microbiome, and imaging datasets, tool suites for single-cell omics and machine learning, and thousands of more tools. Galaxy’s graphical user interface can be used with only a web browser, and there is a programmatic API for performing scripted and automated analyses with Galaxy.</p>
<p><img src="resources/images/05-General_Platforms_files/figure-html/1B4LwuvgA6aUopOHEAbES1Agjy7Ex2IpVAoUIoBFbsq0_gfb2e21ecdc_0_131.png" title="Galaxy can be accessed through a web browser and provides users with access to tools, datasets, computing resources, a graphical user interface (GUI) for users who would like to interact with Galaxy by clicking buttons and using drop-down menus and a programmtic API for users that would like to write code to interact with Galaxy" alt="Galaxy can be accessed through a web browser and provides users with access to tools, datasets, computing resources, a graphical user interface (GUI) for users who would like to interact with Galaxy by clicking buttons and using drop-down menus and a programmtic API for users that would like to write code to interact with Galaxy" width="100%" style="display: block; margin: auto;" /></p>
<p>Galaxy is used daily by thousands of scientists across the world. A vibrant Galaxy community has deployed hundreds of Galaxy servers across the world, including more than 150 public and three large national/international servers in the United States, Europe, and Australia (<a href="https://usegalaxy.org" class="uri">https://usegalaxy.org</a>, <a href="https://usegalaxy.eu" class="uri">https://usegalaxy.eu</a>, <a href="https://usegalaxy.org.au" class="uri">https://usegalaxy.org.au</a>). The three national/international servers have more than 250,000 registered users who execute &gt;500,000 analysis jobs each month. Galaxy has been cited more than 10,000 times with &gt;20% from papers related to cancer. The Galaxy Tool Shed (<a href="https://usegalaxy.org/toolshed" class="uri">https://usegalaxy.org/toolshed</a>) provides a central location where developers can upload tools and visualizations and users can search and install tools and visualizations into any Galaxy server. Galaxy has a large presence in the cancer research community. Galaxy serves as an integration and/or analysis platform for 7 projects in the NCI ITCR program. There is also increasing use of Galaxy in key NIH initiatives such as the NCI Cancer Moonshot Human Tumor Atlas Network (HTAN) and the NHGRI Data Commons, called the AnVIL (<a href="https://anvilproject.org/" class="uri">https://anvilproject.org/</a>).</p>
<p><img src="resources/images/05-General_Platforms_files/figure-html/1B4LwuvgA6aUopOHEAbES1Agjy7Ex2IpVAoUIoBFbsq0_gfb2e21ecdc_0_135.png" title="Map of the 3 Galaxy servers" alt="Map of the 3 Galaxy servers" width="100%" style="display: block; margin: auto;" /></p>
<p>Galaxy’s user interface, accessible via a web browser, provides access to all Galaxy functionality. The main Galaxy interface has three panels: available tools (left), running analyses and viewing data (middle), and a full history of tools run and datasets generated (right).</p>
<p><img src="resources/images/05-General_Platforms_files/figure-html/1B4LwuvgA6aUopOHEAbES1Agjy7Ex2IpVAoUIoBFbsq0_gfb2e21ecdc_0_111.png" title="Galaxy user interface showing the 3 panels: left = tools, middle = analysis and data, and right = history and generated datasets" alt="Galaxy user interface showing the 3 panels: left = tools, middle = analysis and data, and right = history and generated datasets" width="100%" style="display: block; margin: auto;" /></p>
<p>Datasets for analysis in Galaxy can be <strong>uploaded</strong> from a laptop or desktop computer or obtained from public data repositories connected to Galaxy. With Galaxy, complex workflows composed of tens or even hundreds of analysis tools can be created and run. In Galaxy’s workflow interface, tools can be added and connected via a simple drag-and-drop approach.</p>
<p><img src="resources/images/05-General_Platforms_files/figure-html/1B4LwuvgA6aUopOHEAbES1Agjy7Ex2IpVAoUIoBFbsq0_gfb2e21ecdc_0_141.png" title="Galaxy workflow example showing the beginning steps of a workflow" alt="Galaxy workflow example showing the beginning steps of a workflow" width="100%" style="display: block; margin: auto;" /></p>
<p>Galaxy users can share all their work—analysis histories, workflows, and visualizations—via simple URLs that are available to specific colleagues or a link that anyone can access. Galaxy’s user interface is highly scalable. Tens, hundreds, or even thousands of datasets can be grouped into collections and run in parallel using individual tools or multi-tool workflows. In summary, Galaxy is a popular computational workbench with tools and features for a wide variety of data analyses, and it has broad usage in cancer data analysis.</p>
<p>See<a href="https://toolshed.g2.bx.psu.edu/">here</a> for the list of applications supported by Galaxy.</p>
<p>AnVIL?</p>
<p>Terra and atmosphere?</p>
</div>
<div id="cyverse" class="section level3" number="5.0.2">
<h3><span class="header-section-number">5.0.2</span> CyVerse</h3>
<p><a href="https://cyverse.rocks/about">CyVerse</a> is a computing platform or cyberinfastructure similar to Galaxy that also offers computing resources for storing, sharing, and working with data with a graphical interface, as well as an API. Originally called iPlant Collaborative, it was started by a funding from the National Science Foundation (NSF) to support life sciences research, particularly to support ecology, biodiversity, sustainability, and agriculture research. It is led by the University of Arizona, the Texas Advanced Computing Center, and Cold Spring Harbor Laboratory. It offers access to an environment for performing analyses with Jupyter (for Python mostly) and RStudio (for R mostly) and a variety of tools for Genomic data analysis. See <a href="https://cyverse.atlassian.net/wiki/spaces/DEapps/pages/241882146/List+of+Applications">here</a> for a list of applications that are supported by CyVerse. Note that you can also install tools on both platforms. Both CyVerse and Galaxy offer lots of helpful documentation, to help users get started with informatics analyses.</p>
<p>See <a href="https://learning.cyverse.org/en/latest/">here</a> to learn more.</p>
<p><img src="resources/images/05-General_Platforms_files/figure-html/1B4LwuvgA6aUopOHEAbES1Agjy7Ex2IpVAoUIoBFbsq0_gfd56752f25_0_0.png" title="CyVerse graphical interface for performing analyses" alt="CyVerse graphical interface for performing analyses" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="sciserver" class="section level3" number="5.0.3">
<h3><span class="header-section-number">5.0.3</span> SciServer</h3>
<p>SciServer is similar to Galaxy in that it is accessible through a web browser and allows users to store, upload, download, share, and work with data and common tools on the same platform. It was originally built for the astrophysics community (and called SkyServer) but it has now been adapted to be used by scientists of all fields and is indeed used by many in the genomics field. It allows users to use Python and R in environments like Jupyter notebooks and RStudio, and also supports (Structured Query Language) SQL for data querying and management and is built on the use of Docker.</p>
<p>The main idea of SciServer, is based on this premise: “bring the analysis to the data”. It is free to use after users register. However, users can buy extra resources. Users can keep data private or share their data.</p>
<p>As compared to Galaxy, this resources may be better for users with a bit more familiarity with informatics but who require more flexibility, particularly for working with collaborators such as physicists or material scientists as there are more tools supported across disciplines. In addition it also gives users access to very large data sets on Petabyte-scale (note that some of these require special permission to use) and supports developers to create their own web interfaces called SciUIs for particular use cases.</p>
<p>For more information see
<a href="https://reader.elsevier.com/reader/sd/pii/S2213133720300664?token=D3EB0851DC2A36526E6999F0D95DA1E6C41D01E9787B690B2F06F7CE7C192B7141B9FA6BAE5DF1D4BFAD7E402540624E&amp;originRegion=us-east-1&amp;originCreation=20211028203856" class="uri">https://reader.elsevier.com/reader/sd/pii/S2213133720300664?token=D3EB0851DC2A36526E6999F0D95DA1E6C41D01E9787B690B2F06F7CE7C192B7141B9FA6BAE5DF1D4BFAD7E402540624E&amp;originRegion=us-east-1&amp;originCreation=20211028203856</a></p>
<p><img src="resources/images/05-General_Platforms_files/figure-html/1B4LwuvgA6aUopOHEAbES1Agjy7Ex2IpVAoUIoBFbsq0_gfd56752f25_0_13.png" title="The SciServer dashboard allows users to interact with the various resources and tools available through SciServer. Most are available through the command line but some have user interface options. Jupyter Notebooks, RStudio and SciUIs (interfaces created by other SciServer users) can be used to interactively explore and analyze data by users." alt="The SciServer dashboard allows users to interact with the various resources and tools available through SciServer. Most are available through the command line but some have user interface options. Jupyter Notebooks, RStudio and SciUIs (interfaces created by other SciServer users) can be used to interactively explore and analyze data by users." width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="materials-cloud" class="section level3" number="5.0.4">
<h3><span class="header-section-number">5.0.4</span> Materials Cloud</h3>
<p>Another resource that might be of interest to Python users, particular those who collaborate with material scientists, is Materials Cloud. It is designed to promote reproducible work, collaboration, and sharing of resources among scientists, particularly for simulations for the materials science field. Users can share data in a citable way, download data, upload data, share workflows, and perform analyzes.</p>
<p><img src="resources/images/05-General_Platforms_files/figure-html/1B4LwuvgA6aUopOHEAbES1Agjy7Ex2IpVAoUIoBFbsq0_gfd56752f25_0_6.png" title="Materials Cloud resources are based on allowing users to Learn about resources, Work using the resources, Discover aspects about data that is available, Explore data with interactive graphs, and archive to store and share data." alt="Materials Cloud resources are based on allowing users to Learn about resources, Work using the resources, Discover aspects about data that is available, Explore data with interactive graphs, and archive to store and share data." width="100%" style="display: block; margin: auto;" /></p>
<p>This resource uses <a href="https://www.materialscloud.org/work/aiidalab">AiiDAlab</a> as the computing environment for researchers, which is based on <a href="https://www.nature.com/articles/s41597-020-00638-4">AiiDA</a>. According to their website:</p>
<blockquote>
<p>AiiDAlab builds on <a href="https://www.nature.com/articles/s41597-020-00638-4">AiiDA</a> as the computational workflow engine, and the Jupyter environment (notebooks, widgets, …) for writing and sharing apps</p>
</blockquote>
<p>See <a href="https://www.sciencedirect.com/science/article/pii/S092702562030656X?via%3Dihub">here</a> to learn more about how AiiDAlab supports the sharing of scientific workflows, particularly for those that use Python.</p>
<p>To learn more about Materials Cloud, check out this <a href="https://www.nature.com/articles/s41597-020-00637-5">article</a></p>
</div>
<div id="atlas.ti" class="section level3" number="5.0.5">
<h3><span class="header-section-number">5.0.5</span> ATLAS.ti</h3>
<p><a href="https://atlasti.com/">ATLAS.ti</a> is another option but particularly for qualitative analysis. You can use a variety of data types including video, audio, images, surveys, and social media data. A variety of tools, particularly for text data analysis are provided for methods such as <a href="https://en.wikipedia.org/wiki/Sentiment_analysis">sentiment analysis</a>, which is the process of assigning a general tone or feeling to text and <a href="https://en.wikipedia.org/wiki/Named-entity_recognition">named-entity recognition</a>, which is the process of extracting certain characteristics from texts that are what is called a [named entity] or a real-world object - such as a person’s name or address. Such analyses can be helpful for understanding behaviors that might be associated with cancer risk. Although this type of analysis can be performed using R or Python among other coding languages, ATLAS.ti offers a nice graphical user interface to perform these types of analyses s.Furthermore ATLAS.ti offers a great deal of flexibility about such analyses using different data types easily.</p>
<iframe src="https://downloads.atlasti.com/docs/branding/atlasti_brochure_v9_EN_interactive_202110.pdf" width="672" height="400px">
</iframe>
<p>Avocado - want to add in content about price, security, privacy/HIPAA, scalability/file/ram limits for each if possible.</p>

</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="shared-computing-etiquette.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="specific-platforms.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
